File: src/main/kotlin/codeacademy/bookingforum/app/comment/CommentController.java
package codeacademy.bookingforum.app.comment;

import codeacademy.bookingforum.app.configuration.ResponseObject;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.security.access.annotation.Secured;
import org.springframework.web.bind.annotation.*;
import org.springframework.web.context.request.WebRequest;

import java.util.List;

@RestController
@RequestMapping("/api/comment")
public class CommentController {
    @Autowired
    CommentService commentService;

    @Secured({"ROLE_ADMIN","ROLE_USER","ROLE_SELLER"})
    @PostMapping("/create")
    @ResponseStatus(HttpStatus.CREATED)
    public ResponseObject create(@RequestBody CommentDto comment, WebRequest request) {
        return commentService.create(comment, request);
    }

    @GetMapping("/get/list/{id}") // ID of post
    public List<Long> getList(@PathVariable("id") Long id) {
        return commentService.getList(id);
    }

    @GetMapping("/get/{id}") // ID of comment
    public CommentDto getById(@PathVariable("id") Long id) {
        return commentService.getById(id);
    }
}

File: src/main/kotlin/codeacademy/bookingforum/app/comment/CommentController.kt
package codeacademy.bookingforum.app.comment

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.*

@Controller
@RequestMapping("/comment")
class CommentController {

    @Autowired
    private lateinit var service: CommentService

    @GetMapping("/getAllComments")
    fun getComment(): List<CommentDto>? {
        return service.getAllComments()
    }

    @GetMapping("/getComment/{id}")
    fun getCommentById(@PathVariable id: Long): CommentDto? {
        return service.getCommentById(id)
    }

    @PostMapping("/postComment")
    fun postComment(@RequestBody commentDto: CommentDto?) {
        return service.postComment(commentDto)
    }

    @PutMapping("/updateComment/{id}")
    fun updatePost(@RequestBody commentDto: CommentDto?, @PathVariable id: Long) {
        return service.updateComment(id, commentDto)
    }

    @DeleteMapping("/deleteComment/{id}")
    fun deleteComment(@PathVariable id: Long): String {
        return service.deleteComment(id)
    }

}